### PR TITLE
Add Agent hardware and concurrency clarifications

### PR DESCRIPTION
**Pull request by mikegreen**
## Labels

- [ ] inaccuracy
- [x] clarification
- [ ] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout

### CODE REVIEW

File: content/source/docs/cloud/agents/index.html.md
Comment (vaficionado) [inline content/source/docs/cloud/agents/index.html.md:28]:
```suggestion
The instance running the agent will have varying resource requirements depending on the workspace being run. Memory, CPU and disk space should be monitored on the agent instance and adjusted accordingly. The specs below are provided as a reference. Depending on the type of environment you deploy to, the name of your instance may vary.
```

File: content/source/docs/cloud/agents/index.html.md
Comment (vaficionado) [inline content/source/docs/cloud/agents/index.html.md:31]:
```suggestion
  * Each run requires the agent to temporarily store local copies of the tarred repository, extracted repository, state file, any providers or modules, and the Terraform binary itself.
```

File: content/source/docs/cloud/agents/index.html.md
Comment (vaficionado) [inline content/source/docs/cloud/agents/index.html.md:216]:
```suggestion
Each agent process will run a single Terraform run at a time. Multiple agent processes can be concurrently run on a single instance, license limit permitting. 
```

**Review by vaficionado — CHANGES_REQUESTED** [review]
Made some tweaks for clarity, formatting, other things. Also think @glasner should have a look to be sure we are aligned for TFE as well

File: content/source/docs/cloud/agents/index.html.md
Comment (vaficionado) [inline content/source/docs/cloud/agents/index.html.md:28]:
```suggestion
The instance running the agent will have varying resource requirements depending on the workspace being run. Memory, CPU and disk space should be monitored on the agent instance and adjusted accordingly. The specs below are provided as a reference. Depending on the type of environment you deploy to, the name of your instance type may vary.
```

Comment (vaficionado) [reply content/source/docs/cloud/agents/index.html.md:28]: I'm not sure if I missed this word in the last review, but it's important here

**Review by vaficionado — COMMENTED** [review]
This reads clearly to me now, but would still like someone from Eng to weigh in and ensure the hard numbers provided check out.

**Comment by glasner** [comment]
Spoke to Mike in Slack, but for transparency... I think we should avoid talking about "instance type" since one of the most common use cases is deploying to k8s. I'm at a loss for what word to use instead though :)

File: content/source/docs/cloud/agents/index.html.md
Comment (glasner) [inline content/source/docs/cloud/agents/index.html.md:33]: Unless we have proof around the number of CPU cores, I would separate it from the min requirements as a note that calls out Terraform will spread out across cores if they are made available.

Comment (mikegreen) [reply content/source/docs/cloud/agents/index.html.md:33]: My logic on that was with 1 core that the agent, terraform main proc, forked procs, and the providers all compete and thrash for CPU and things slowed down a bit when I was playing locally (non cloud stuff).
I think we can remove it - if run on 1 core it works, just slower which should be obvious to most.

**Comment by glasner** [comment]
I like the idea of calling it out to avoid questions, it just doesn't feel
like a minimum requirement

**Comment by laurapacilio** [comment]
Hi all - reviving this! Is this still something we'd want to get merged in?

**Comment by vaficionado** [comment]
> Hi all - reviving this! Is this still something we'd want to get merged in?

I think so, yeah! If it's been reviewed as technically accurate now. I think Jordan still had some open concerns

**Comment by laurapacilio** [comment]
@glasner Let us know what you think when you have a chance. Thank you! :-)

**Comment by mikegreen** [comment]
Removed the debatable 2-cores.. i think we're good to go, now?

File: content/source/docs/cloud/agents/index.html.md
Comment (laurapacilio) [inline content/source/docs/cloud/agents/index.html.md:28]: Just thinking about active voice, flow (I want the specs sentence to come right before the specs), and concision. Please make sure I didn't materially change the meaning!

```suggestion
The host running the agent will have varying resource requirements depending on the workspace. A host can be a dedicated or shared cloud instance, virtual machine, bare metal server, or a container. You should monitor and adjust memory, CPU, and disk space based on each workspace's usage and performance. The name of your instance type may vary depending on your deployment environment. 

The specs below are provided as a reference. 
```

**Review by laurapacilio — APPROVED** [review]
Just reviewed for style and flow - thank you!

**Comment by mikegreen** [comment]
Hey @laurapacilio I can technically merge this.. is it ok to do so? Or is there a desire to have the TF docs team do it?

**Comment by laurapacilio** [comment]
Hey - thank you for asking! My take is: I've approved (thank you for taking my edits!) and so it's fine from where I stand. As long as it's technically correct and eng has approved, you can feel free to merge if you have permissions! I see there's still a pending review, so I'll leave that up to you to resolve.

**Pull request merged**
